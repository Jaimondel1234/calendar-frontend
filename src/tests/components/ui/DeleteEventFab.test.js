import React from "react";

import "@testing-library/jest-dom";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { DeleteEventFab } from "../../../components/ui/DeleteEventFab";
import { eventStartDelete } from "../../../actions/events";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};

const store = mockStore(initState);

store.dispatch = jest.fn();

jest.mock("../../../actions/events", () => ({
  eventStartDelete: jest.fn(),
}));
let token = "";

const wrapper = mount(
  <Provider store={store}>
    <DeleteEventFab />
  </Provider>
);

describe("Pruebas en <DeleteEventFab />", () => {
  test("debe mostrarseccorrectamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("debe de llamar el eventStartDelete al hacer click", () => {
    wrapper.find("button").prop("onClick")();
    expect(eventStartDelete).toHaveBeenCalled();
  });
});
