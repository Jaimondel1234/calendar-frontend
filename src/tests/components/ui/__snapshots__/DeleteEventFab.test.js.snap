// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DeleteEventFab /> debe mostrarseccorrectamente 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  >
    <span>
       Borrar evento
    </span>
  </i>
</button>
`;
